### PR TITLE
Add Julia v1.10.0

### PR DESCRIPTION
**Pull request by giordano**
Companion PR to https://github.com/compiler-explorer/infra/pull/1201.

### CODE REVIEW

File: etc/config/julia.amazon.properties [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+defaultCompiler=julia_1_10_0
 compilerType=julia
 
-group.julia.compilers=julia_1_9_3:julia_1_8_5:julia_1_7_3
+group.julia.compilers=julia_1_10_0:julia_1_9_3:julia_1_8_5:julia_1_7_3
 group.julia.isSemVer=true
 group.julia.baseName=Julia
+compiler.julia_1_10_0.exe=/opt/compiler-explorer/julia-1.10.0/bin/julia
+compiler.julia_1_10_0.semver=1.10.0
+compiler.julia_1_10_0.alias=julia_1_10_0
```
Comment (mattgodbolt) [inline etc/config/julia.amazon.properties:11]: This alias is unnecessary; I will remove (likewise the one below)

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
I don't know what's supposed to happen but when I tested I get:

```
ERROR: LoadError: MethodError: no method matching iterate(::Core.MethodInstance)

Closest candidates are:
  iterate(!Matched::LibGit2.GitRebase)
   @ LibGit2 /opt/compiler-explorer/julia-1.10.0/share/julia/stdlib/v1.10/LibGit2/src/rebase.jl:48
  iterate(!Matched::LibGit2.GitRebase, !Matched::Any)
   @ LibGit2 /opt/compiler-explorer/julia-1.10.0/share/julia/stdlib/v1.10/LibGit2/src/rebase.jl:48
  iterate(!Matched::Base.MethodSpecializations)
   @ Base reflection.jl:1148
  ...

Stacktrace:
 [1] top-level scope
   @ /infra/.deploy/etc/scripts/julia_wrapper.jl:83
in expression starting at /infra/.deploy/etc/scripts/julia_wrapper.jl:75
Compiler returned: 1
```

**Comment by mattgodbolt** [comment]
I'm going to push on and deploy this to our production systems (as there's a lot of other changes too) - let me know if we can help you debug this....I don't know enough about Julia or the compiler to know what's going on here.

**Comment by giordano** [comment]
Sorry for not having tested it earlier, thanks for catching the issue!  #5963 seems to fix it for me.

**Pull request merged**